### DISCUSSION TITLE
Are there any plans to expand the API?

### DISCUSSION THREAD

**Discussion by nearueng** [postalserver/postal#2362] [Development discussions]
Hi there,

We're currently exploring some options to power our sales engagement platform.

Ideally, we'd want to be able to create organizations, mail servers, add domains, get SPF, DKIM, DMARC records, etc., all through an API.

Is this something that is on the roadmap? Apologies if I missed it somewhere.

**Comment by willpower232** [comment]
@adamcooke was working on an API v2 but I'm not sure all the planned functionality it covers.

**Reply by nearueng** [reply]
@willpower232 oh okay. This is a super cool project! 

@adamcooke is there any roadmap published anywhere? 

I don't mean to sound ungrateful with all the hard work that's been done on this project, and an API is not a make or break requirement for us right now.

Thanks again to the Krystal team for all the great work on this!

**Reply by adamcooke** [reply]
I'm not making any promises on timescales but we do have a use-case ourselves for a complete API for Postal in the not too distant future so it is on the roadmap.

**Reply by SimonBiggs** [reply]
@adamcooke, do you suspect that the API might include solving the following issue?

https://github.com/postalserver/postal/issues/1376

**Reply by arthurzenika** [reply]
@adamcooke would that API include the ability to manipulate _Routes_ ?

**Comment by rduffner** [comment]
I guess there is no news here?
Without an API, it's sort of useless for us.
